File: MojStomatolog/MojStomatolog.Models/Core/Order.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace MojStomatolog.Models.Core
{
    public class Order
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public int PaymentId { get; set; }
        public DateTime OrderDate { get; set; }
        public decimal TotalAmount { get; set; }
        public int Status { get; set; }

        public HashSet<OrderItem> OrderItems { get; set; } = [];

        [ForeignKey("UserId")]
        public User User { get; set; } = null!;

        [ForeignKey("PaymentId")]
        public Payment Payment { get; set; } = null!;
    }
}

File: MojStomatolog/UI/mojstomatolog_desktop/lib/models/order.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:mojstomatolog_desktop/models/order_item.dart';
import 'package:mojstomatolog_desktop/models/payment.dart';
import 'package:mojstomatolog_desktop/models/user.dart';

part 'order.g.dart';

@JsonSerializable()
class Order {
  int? id;
  int? userId;
  int? paymentId;
  int? quantity;
  DateTime? orderDate;
  double? totalAmount;
  int? status;
  List<OrderItem>? orderItems;
  Payment? payment;
  User? user;

  Order();

  factory Order.fromJson(Map<String, dynamic> json) => _$OrderFromJson(json);
  Map<String, dynamic> toJson() => _$OrderToJson(this);
}
